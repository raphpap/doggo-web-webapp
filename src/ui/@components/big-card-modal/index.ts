export * from './big-card-modal';
export {default} from './big-card-modal';
