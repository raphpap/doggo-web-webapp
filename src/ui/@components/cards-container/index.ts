export * from './card-container';
export {default} from './card-container';
