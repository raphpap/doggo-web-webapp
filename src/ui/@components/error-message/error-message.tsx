// Vendor
import React from 'react';
import styled from 'react-emotion';

// Theme
import theme from 'doggo-web-webapp/theme';

interface Props {
  message: string;
}

const Container = styled.div`
  display: flex;
  opacity: 0.8;
  align-items: center;
  margin: ${theme.margin.unit * 1.5}px 0 ${theme.margin.unit}px;
  font-size: 13px;
  line-height: 15px;
  color: ${theme.colors.white.plain};
`;

export const ErrorMessage: React.SFC<Props> = ({message}) => (
  <Container>
    <span>{message}</span>
  </Container>
);

export default ErrorMessage;
