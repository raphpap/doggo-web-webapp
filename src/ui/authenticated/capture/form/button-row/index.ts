export * from './button-row';
export {default} from './button-row';
