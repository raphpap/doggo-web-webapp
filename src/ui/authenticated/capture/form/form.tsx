// Vendor
import React, {ChangeEventHandler, FormEventHandler} from 'react';

// Shared components
import ErrorMessage from 'doggo/ui/@components/error-message';

// Components
import Button from './button';
import ButtonRow from './button-row';
import Image from './image';
import TextField from './text-field';

// Types
interface Props {
  imageSrc: string;
  onClose: () => void;
  onSubmit: (cardName: string, imageSrc: string) => void;
}

interface State {
  cardName: string;
  hasError: boolean;
  isDirty: boolean;
}

const cardNameHasError = (cardName: string) => cardName === '';

export class Form extends React.Component<Props, State> {
  public readonly state: State = {
    cardName: '',
    hasError: false,
    isDirty: false
  };

  public render() {
    const {imageSrc, onClose} = this.props;
    const {cardName, hasError, isDirty} = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <TextField
          id="card-name"
          error={!isDirty && hasError}
          placeholder="Card Name"
          type="text"
          value={cardName}
          onChange={this.handleCardName}
        />

        {!isDirty && hasError && (
          <ErrorMessage message="Enter a name for your card" />
        )}

        <Image src={imageSrc} />

        <ButtonRow>
          <Button onClick={onClose}>Cancel</Button>
          <Button type="submit">Confirm</Button>
        </ButtonRow>
      </form>
    );
  }

  private handleSubmit: FormEventHandler<HTMLFormElement> = async event => {
    event.preventDefault();
    const {onSubmit, imageSrc} = this.props;
    const {cardName} = this.state;

    this.setState({
      hasError: cardNameHasError(cardName),
      isDirty: false
    });

    if (!cardNameHasError(cardName)) {
      onSubmit(cardName, imageSrc);
    }
  };

  private handleCardName: ChangeEventHandler<HTMLInputElement> = event => {
    const cardName = event.target.value;

    this.setState({
      cardName,
      hasError: cardNameHasError(cardName),
      isDirty: true
    });
  };
}

export default Form;
