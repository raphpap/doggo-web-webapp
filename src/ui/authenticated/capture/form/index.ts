export * from './form';
export {default} from './form';
