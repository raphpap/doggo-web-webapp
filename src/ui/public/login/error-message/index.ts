export * from './error-message';
export {default} from './error-message';
