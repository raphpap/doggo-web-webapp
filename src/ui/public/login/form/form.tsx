// Vendor
import styled from 'react-emotion';

// Theme
import theme from 'doggo-web-webapp/theme';

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: ${theme.margin.unit * 2.5}px;
`;

export default Form;
