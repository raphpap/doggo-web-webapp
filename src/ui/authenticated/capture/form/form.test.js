// Vendor
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// Shared Components
import {ErrorMessage} from 'doggo/ui/@components/error-message';

// Components
import {Form} from './form';
import TextField from './text-field';

// Mocks
import {mockContextWithoutData, mockLoggedIn} from 'doggo/test-mocks';

describe('<Form />', () => {
  let wrapper;
  let context;
  let imageSrc = "image_src";
  let onClose = jest.fn();
  let onSubmit = jest.fn();

  let props = {
    imageSrc,
    onClose,
    onSubmit
  };

  beforeEach(() => {
    wrapper = shallow(<Form {...props} />);
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('shows the errors when the submit fails', () => {
    wrapper.setState({hasError: true});

    const errorMessage = wrapper.find(ErrorMessage);
    const textField = wrapper.find(TextField);

    expect(textField.props().error).toBe(true);
    expect(errorMessage).toHaveLength(1);
  });

  it('does not call onSubmit when submitting the form and the card name is not valid', () => {
    const fakeEvent = { preventDefault: jest.fn() };
    wrapper.setState({cardName: '   '});
    const form = wrapper.find('form');
    form.simulate('submit', fakeEvent);

    expect(fakeEvent.preventDefault).toBeCalled();
    expect(props.onSubmit).not.toBeCalled();
  });

  it('calls onSubmit when submitting the form and the card name is valid', () => {
    const fakeEvent = { preventDefault: jest.fn() };
    wrapper.setState({cardName: 'card_name'});
    const form = wrapper.find('form');
    form.simulate('submit', fakeEvent);

    expect(fakeEvent.preventDefault).toBeCalled();
    expect(props.onSubmit).toBeCalled();
  });
});
