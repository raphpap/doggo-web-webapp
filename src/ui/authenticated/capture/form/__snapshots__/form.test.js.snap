// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    error={false}
    id="card-name"
    onChange={[Function]}
    placeholder="Card Name"
    type="text"
    value=""
  />
  <Image
    src="image_src"
  />
  <ButtonRow>
    <Button
      onClick={[MockFunction]}
    >
      Cancel
    </Button>
    <Button
      type="submit"
    >
      Confirm
    </Button>
  </ButtonRow>
</form>
`;
